setup.py

```diff
@@ -13,6 +15,15 @@
         "mypy-extensions>=0.3"
     ]
 
+extensions = [Extension(
+    name="solving._multiparty_solving", 
+    sources=["anonlink/solving/_multiparty_solving.pyx",
+             "anonlink/solving/_multiparty_solving_inner.cpp"],
+    language="c++",
+    extra_compile_args=["-std=c++11"],
```

Comment:
You should ensure that the `Extension` mechanism specifies `-DNDEBUG` when compiling; if it doesn't, you should specify it here. You might also consider how to make sure `-DNDEBUG` is *not* set when running tests (or otherwise when might want the `assert`ions to be checked).

Reply:
Good thought. I've explicitly added `NDEBUG` to the defined macros.

I would also like to make it an option to install without `NDEBUG`, but this seems difficult to do in setup.py and I don't have the patience.